ttnn/ttnn/multi_device.py

```diff
@@ -443,7 +446,30 @@ def __init__(self, mesh_device: MeshDevice):
         self.mesh_device = mesh_device
 
     def compose(self, tensor: ttnn.Tensor) -> List["torch.Tensor"]:
-        return [ttnn.to_torch(tt_input_tensor) for tt_input_tensor in ttnn.get_device_tensors(tensor)]
+        return [
+            ttnn.to_torch(tt_input_tensor, mesh_composer=None) for tt_input_tensor in ttnn.get_device_tensors(tensor)
+        ]
+
+
+@contextlib.contextmanager
+def distribute(mesh_mapper: TensorToMesh = None, mesh_composer: MeshToTensor = None):
+    if mesh_mapper is None and mesh_composer is None:
+        raise ValueError("Either mesh_mapper or mesh_composer must be provided.")
+
+    _original_to_torch = ttnn.to_torch
+    _original_from_torch = ttnn.from_torch
+
+    try:
+        if mesh_mapper:
+            ttnn.from_torch = functools.partial(_original_from_torch, mesh_mapper=mesh_mapper)
+        if mesh_composer:
+            ttnn.to_torch = functools.partial(_original_to_torch, mesh_composer=mesh_composer)
+        yield
+
+    finally:
+        # Restore the original functions
```

Comment:
Pretty creative. I think it looks nice. Maybe a little surprising to the end-user. End user might be surprised to see no from/to_torch calls in the visualizer. But it seems worthy the benefit and user will learn this fast.

Reply:
the user should still be able to visualize these `from_torch`/`to_torch` invocations. The only thing that should change are the default kwargs.